### ISSUE TITLE
Improve how large assets (music) are handled

### ISSUE THREAD

**Issue by IntegratedQuantum** [PixelGuys/Cubyz#763]
#### There are some problems with the current system:
- Right now they are redownloaded every time the game is updated using [the easy way](https://github.com/PixelGuys/Cubyz?tab=readme-ov-file#the-easy-way-no-tools-needed).
- Either way they are downloaded before launching the game (→ more waiting for first-time player).
- For server side assets, they are downloaded from the server (→ unnecessary bandwidth usage on join)
- For server side assets, they are redownloaded every time (→ not that important, since there are other fixes)

#### All of them could be solved like this:
- music is no longer stored in the assets, instead the assets only contain download links to the music
- music is stored in a global cache, which is checked first before trying to download a song

#### The disadvantages of this are obvious:
- addon creators need to host the music somewhere, instead of just bundling them in with the other assets
- music of these addon would be prone to link rot and are likely to break at some point in the future.

I'm not sure how big of a deal these problems are.
Publicly hosting files like music today is easier than ever and it's often free.
And if there is no community for an addon to fix the link rot, then it doesn't seem like it would be worth caring about anyways.

One possible solution for addons would be to just allow the old behavior using local links with relative file paths.

@ikabod-kee I would like to hear your opinion on this.

**Comment by IntegratedQuantum** [comment]
> I think a good option could be to have a Cubyz updater that updates Cubyz to the latest version, so you don't have to download the game repeatedly. The only reason having this many music files is a problem is because it adds onto compilation time.

Yeah that's planned with the launcher, but the launcher should not handle downloading the game assets, because that would make regular development (without the launcher) more difficult.

Furthermore if the launcher downloads the music, then it must happen before the game starts, what I'm suggesting is to download it while the game is already running. Downloading 200 MB can take quite a while on bad connections.

> For servers assets, a link could work, yeah, but you could also solve this by providing the add-on to the player to pre-install before joining, or compressing the music files.

Yes, that is an option, but I prefer the per-world asset model that has been established, because it's zero work for the player.
Also note that this doesn't just apply to the server, but any world with these addons. If the world file contains 100 MB of music, then that is going to bloat the saves folder.

**Comment by IntegratedQuantum** [comment]
No, I think I'll get this done soon anyways.

**Comment by IntegratedQuantum** [comment]
Since I wasn't sure if we really should download these individually at runtime (and maybe also the fact that I couldn't get the http client to work), I decided to just use Zig's compile-time caching system for now.

A further decision has to be made here, but I'm postponing this after #1483 since new music is now again allowed to be added to the new repo https://github.com/PixelGuys/Cubyz-Assets.

> No, I think I'll get this done soon anyways.

Surely 8 months still counts as soon, right?

**Comment by IntegratedQuantum** [comment]
no more duplicate webhooks

**Comment by codemob-dev** [comment]
What about https://github.com/ikabod-kee/CubyzMusic/tree/main?

**Comment by IntegratedQuantum** [comment]
@ikabod-kee could you make a PR that adds all the music to the new repo?
